python/res/enkf/site_config.py

```diff
@@ -147,7 +147,10 @@ def __eq__(self, other):
         self_job_list = self.get_installed_jobs()
         other_job_list = other.get_installed_jobs()
 
-        if other_job_list.getAvailableJobNames() != self_job_list.getAvailableJobNames():
+        if len(other_job_list.getAvailableJobNames()) != len(self_job_list.getAvailableJobNames()):
+            return False
+
+        if not all(x in other_job_list.getAvailableJobNames() for x in self_job_list.getAvailableJobNames()):
```

Comment:
This does not check for equality, it checks if `self_job_list` is a subset of `other_job_list`. What about checking for `set(other_job_list.getAvailableJobNames()) != set(self_job_list.getAvailableJobNames())`